tools/patchwork_integration_test.sh

```diff
@@ -0,0 +1,113 @@
+#! /bin/bash
+
+# Author:
+#   Rohit Sarkar <rohitsarkar5398@gmail.com>
+#
+# This work is licensed under the terms of the GNU GPL, version 2.  See
+# the COPYING file in the top-level directory.
+
+set -e
+USAGE=$(cat <<-END
+	$ ./patchwork_integration_test.sh <path-to-public-inbox> <timestamp>
+	Only mails in public inbox since the given <timestamp> will be considered
+	END
+	)
+if [ $# -eq 0 ]
+then
+	echo -e "$USAGE"
+	exit 1
+fi
+
+HOME="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+PUBIN_PATH=$1
+SINCE=$2
+MBOX=test-mbox.mbox
+PROJECT=test-patchwork
+INITIAL_IMPORT_SIZE=50 # Percentage of mbox to be imported as a raw mbox
+INITIAL_IMPORT_MBOX=initial.mbox
+
+clean_up() {
+	1>/dev/null 2>&1 docker rm -f pasta-patchwork
+}
+
+trap clean_up EXIT
+
+get_config() {
+	# Get a minimal config for PaStA
+	# For api imports we keep the intial archive field empty
+	if [ $1 = "mbox" ]
+	then
+		initial_archive=$INITIAL_IMPORT_MBOX
+	else
+		initial_archive=""
+	fi
+	config=$(cat <<-END
+		[PaStA]
+		MODE = "mbox"
+		UPSTREAM = "HEAD^..HEAD"
+
+		[mbox]
+		[mbox.patchwork]
+		url='http://$container_addr:8000/api/'
+		projects = [{id=$project_id, initial_archive="$initial_archive", list_email="test@test.com"}]
+		END
+	)
+}
+
+cd "$HOME"
+
+echo "Parsing pubin..."
+./pubin2mbox.sh $PUBIN_PATH $SINCE > $MBOX
+if [ ! -d patchwork ]
+then
+	git clone https://github.com/rsarky/patchwork.git
+fi
+
+cp ./$MBOX patchwork/
+
+cd patchwork
+git checkout pasta-patchwork-integration
+rm -f *.tar
+echo "UID=$UID" > .env
+echo "GID=`id -g`" >> .env
+docker-compose build
+docker-compose run --rm web ./pasta-setup.sh $MBOX $PROJECT
```

Comment:
Please leave a comment what this comment is meant to do. E.g.: Add $MBOX to patchwork (parsearchive), and export it again so that the result contains x-patch-id hdrs.